ytmusicapi/parsers/songs.py

```diff
@@ -21,9 +21,18 @@ def parse_song_artists_runs(runs: JsonList) -> JsonList:
     return artists
 
 
-def parse_song_runs(runs: JsonList) -> JsonDict:
+def parse_song_runs(runs: JsonList, skip_type_spec: bool = False) -> JsonDict:
+    """
+    :param skip_type_spec: ignore data preceding artist text runs, which may be a type specifier like "Song", "Single", or "Album"
+    """
+
     parsed: JsonDict = {"artists": []}
-    for i, run in enumerate(runs):
+
+    for i, run in reversed(list(enumerate(runs))):
```

Comment:
I don't think we should reverse the list in general. 

I suspect it is done because you want to check the last item and abort. I would suggest putting that check before the loop.

Reply:
That looks good to me, go ahead